### PR TITLE
[TASK] Rework chapter about the runTests.sh and build/testing mechanism

### PR DESCRIPTION
**Pull request by garvinhicking**

### CODE REVIEW

File: Documentation/HandlingAPatch/CherryPick.rst [hunk 1]
```diff
@@ -51,16 +51,20 @@ to cherry-pick it from the review system into your local git repository.
 
 4. Clean up your local repository
 
-   Save your local changes beforehand, if you have any.
-   
+   Save your local changes beforehand, if you have any. Otherwise the
+   `git reset`` performed after that would delete local changes, which
```
Comment (garvinhicking) [inline Documentation/HandlingAPatch/CherryPick.rst:55]:
```suggestion
   `git reset` performed after that would delete local changes, which
```

**Review by sbuerk — CHANGES_REQUESTED** [review]

File: Documentation/Testing/CoreTesting.rst [hunk 2]
```diff
@@ -80,27 +81,25 @@ Overview
...
+will be pulled before continuing. These are the dependent images needed to execute certain jobs. For instance,
+a container providing the specific PHP-version may be fetched. The same containers are used for the
+TYPO3 CI GitLab Pipeline, even utilizing the same `runTests.sh` script. What's impressive is that
+you can locally run the same tests like a fully-fledged CI server..
+
+The GitLab CI Pipeline is maintained through the Ansible infrastructure found on
+`https://git.typo3.org/typo3/CI/testing-infrastructure/-/tree/main/ansible?ref_type=heads`__, and the Pipeline
+itself is set up through `https://github.com/TYPO3/typo3/tree/main/Build/gitlab-ci/`__.
+
+Compared to your local execution tt's just that the combinations of tests
```
[the remark this answers is not in the document]
Comment (garvinhicking) [reply Documentation/Testing/CoreTesting.rst:97]: ack, done

File: Documentation/Testing/CoreTesting.rst [hunk 3]
```diff
@@ -138,7 +138,7 @@ Let's pick a :file:`runTests.sh` example and have a closer look:
 The command asks :file:`runTests.sh` to execute the "functional" test suite `-s functional` and to not execute all
 available tests but only those within `typo3/sysext/core/Tests/Functional/Authentication/`. The script first
 starts the containers it needs: Redis, memcached (previosuly also MariaDB by default, which is now using
-SQLite instead, due to less depndencies). All in one network. It then then starts a PHP 8.0 container and calls
+SQLite instead, due to less depndencies). All in one network. It then then starts a PHP 8.2 container and calls
```
Comment (froemken) [inline Documentation/Testing/CoreTesting.rst:141]:
```suggestion
SQLite instead, due to less dependencies). All in one network. It then starts a PHP 8.2 container and calls
```

File: Documentation/Testing/CoreTesting.rst [hunk 4]
```diff
@@ -152,11 +137,12 @@ Let's pick a :file:`runTests.sh` example and have a closer look:
 
 The command asks :file:`runTests.sh` to execute the "functional" test suite `-s functional` and to not execute all
 available tests but only those within `typo3/sysext/core/Tests/Functional/Authentication/`. The script first
-starts the containers it needs: Redis, memcached and a MariaDB. All in one network. It then waits until
-the MariaDB container opens its database port, then starts a PHP 8.0 container and calls phpunit to execute
-the tests. phpunit executes only one test in this case, that one is green. The containers and networks are then
-removed again. Note the exit code of :file:`runTests.sh` (`echo $?`) is identical to the exit code of the phpunit
-call: If phpunit reports green, :file:`runTests.sh` returns 0, and if phpunit is red, the exit code would be non zero.
+starts the containers it needs: Redis, memcached (previosuly also MariaDB by default, which is now using
```
Comment (froemken) [inline Documentation/Testing/CoreTesting.rst:140]:
```suggestion
starts the containers it needs: Redis, memcached (previously also MariaDB by default, which is now using
```

File: Documentation/Testing/CoreTesting.rst [hunk 5]
```diff
@@ -216,9 +205,9 @@ To speed up test execution, the PHP extension `xdebug` is not usually loaded.
 However, to allow debugging tests and system under tests, it is possible to
 activate xdebug and send debug output to a local IDE. We'll use PhpStorm for this example.
 
-Let's verify our PhpStorm debug settings first. Go to File > Settings > Languages & Frameworks > PHP
-> Debug. Make sure "Can accept external connections" is enabled, remember the port if it is not the
-default port(9000) and also raise "Max. simultaneous connections" to two or three. Note remote debugging
+Let's verify our PhpStorm debug settings first. Go to :guilabel:`File > Settings > Languages & Frameworks > PHP
```
Comment (froemken) [inline Documentation/Testing/CoreTesting.rst:208]: Is that a typo `&_`? Don't know that writing.

File: Documentation/Testing/CoreTesting.rst [hunk 6]
```diff
@@ -171,8 +157,8 @@ are not valid:
 ..  code-block:: shell
 
     lolli@apoc /var/www/local/cms/Web $ Build/Scripts/runTests.sh -h
-    TYPO3 Core test runner. Execute acceptance, unit, functional and other test suites in
-    a docker based test environment. Handles execution of single test files, sending
+    TYPO3 core test runner. Execute acceptance, unit, functional and other test suites in
```
Comment (froemken) [inline Documentation/Testing/CoreTesting.rst:160]:
```suggestion
    TYPO3 Core test runner. Execute acceptance, unit, functional and other test suites in
```

File: Documentation/Testing/CoreTesting.rst (hunk 5 above)
Comment (garvinhicking) [reply Documentation/Testing/CoreTesting.rst:208]: That was just an error, thanks!

**Review by sbuerk — APPROVED** [review]

**Pull request merged**